pkg/flow/internal/controller/loader.go

```diff
@@ -507,6 +535,18 @@ func (l *Loader) wireGraphEdges(g *dag.Graph) diag.Diagnostics {
 
 				g.AddEdge(dag.Edge{From: n, To: dep})
 			}
+		case *DeclareNode:
+			// Although they do nothing on evaluation, DeclareNodes are wired
+			// to detect cyclic dependencies. If a declare "a" block contains an instance
+			// of a declare "b" which contains an instance of the declare "a", both DeclareNodes
+			// will depend on each others, creating a cycle in the graph which will be detected later.
```

Comment:
This comment is fine, but it would be easier to understand with an inline code example; I'd recommend writing one to help illustrate the cyclic dependency you're describing.